app/Http/Controllers/Bible/BibleFileSetsController.php

```diff
@@ -260,22 +269,30 @@ function () use ($fileset_id, $book_id, $chapter_id, $verse_start, $verse_end) {
      * )
      *
      * @param string|null $fileset_url_param
+     * @param string|null $book_url_param
      *
      * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View|mixed
      * @throws \Exception
      */
     public function showBulk(
-        $fileset_url_param = null,
+        $fileset_url_param,
+        $book_url_param = null,
         $cache_key = 'bible_filesets_show_bulk'
     ) {
-        $fileset_id = checkParam('dam_id|fileset_id', true, $fileset_url_param);
-        $cache_params = [$this->v, $fileset_id];
+        $fileset_id  = checkParam('dam_id|fileset_id', true, $fileset_url_param);
+        $book_id     = checkParam('book_id', false, $book_url_param);
+        $cache_params = [$this->v, $fileset_id, $book_id];
+        $limit        = (int) (checkParam('limit') ?? 5000);
```

Comment:
ensure 5000 is also the minimum limit in the case that limit is provided as a parameter